app/components/dropdown.js

```diff
@@ -11,37 +14,63 @@ export default Component.extend({
 		'hasDarkShadow:wds-has-dark-shadow',
 		'isLevel2:wds-dropdown-level-2:wds-dropdown',
 		'isTouchDevice:wds-is-touch-device',
+		'isFlipped:wds-is-flipped',
 	],
 
 	isClicked: false,
 	isLevel2: false,
+	isFlipped: false,
+	canFlip: false,
 	isTouchDevice: isTouchDevice,
 
 	isActive: or('dropdownExpanded', 'isClicked'),
 
+	contentElement: computed('element', function () {
+		return this.element && this.element.querySelector('.wds-dropdown__content');
+	}),
+
 	init() {
 		this._super(...arguments);
 
-		if (this.mouseEnter) {
-			this.mouseEnter = this.mouseEnter.bind(this);
-		}
-		this.mouseLeave = this.mouseLeave.bind(this);
+		this._mouseEnter = this._mouseEnter.bind(this);
+		this._mouseLeave = this._mouseLeave.bind(this);
 	},
 
 	didInsertElement() {
 		this._super(...arguments);
 
-		if (this.mouseEnter) {
-			this.element.addEventListener('mouseenter', this.mouseEnter);
-		}
-
-		this.element.addEventListener('mouseleave', this.mouseLeave);
+		this.element.addEventListener('mouseenter', this._mouseEnter);
+		this.element.addEventListener('mouseleave', this._mouseLeave);
 	},
 
-	mouseLeave() {
+	_mouseLeave() {
 		if (this.isTouchDevice) {
 			this.set('isClicked', false);
 		}
+
+		if (this.canFlip && !this.isLevel2) {
+			this.set('isFlipped', false);
+		}
+
+		if (this.mouseLeave) {
+			this.mouseLeave(...arguments);
+		}
+	},
+
+	_mouseEnter() {
+		if (this.canFlip && !this.isLevel2 && this.contentElement) {
+			const contentElementBoundingRect = this.contentElement.getBoundingClientRect();
+
+			if (contentElementBoundingRect.bottom > getViewportSize().height) {
```

Comment:
`this.set('isFlipped', contentElementBoundingRect.bottom > getViewportSize().height);` ;P?